sqlmesh/core/renderer.py

```diff
@@ -68,6 +68,13 @@ def __init__(
         self._cache: t.List[t.Optional[exp.Expression]] = []
         self._model_fqn = model_fqn
         self._optimize_query_flag = optimize_query is not False
+        # Cache of the macro evaluator to be able to reuse it outside the render method
+        self._macro_evaluator: t.Optional[MacroEvaluator] = None
+
+    @property
+    def macro_evaluator(self) -> t.Optional[MacroEvaluator]:
+        """Returns the cached macro evaluator from the last render operation."""
+        return self._macro_evaluator
```

Comment:
This doesn't feel like a robust way of doing this. Can we instead have a method that constructs the MacroEvaluator instance? And then call it externally AND in `_render`?